### DISCUSSION TITLE
Should inline object declarations be valid?

### DISCUSSION THREAD

**Discussion by remy90** [ChristopheBougere/asl-validator#152] [General]
I've not managed to find much in asl-validator or the aws docs on whether something like the following should be considered legit:
```json
{
     "context.$": "States.JsonMerge($.context, { \"region\": \"FR\"}, false)",
}
```
But it appears to be providing a generic error message:
SCHEMA_VALIDATION_FAILED: Name/Parameters/Input/context.$ is invalid. must match format "asl_payload_template"

Is this actually invalid?

**Comment by massfords** [comment]
It's not clear from the AWS docs if they support inline JSON objects. See their documentation [here](https://docs.aws.amazon.com/step-functions/latest/dg/intrinsic-functions.html#asl-intrsc-func-json-manipulate).

The examples they provide show referencing the arguments to the function from the state as opposed to providing an inline object. 

I believe I coded the rules for the [asl-path-validator](https://github.com/massfords/asl-path-validator) lib to only accept JSON path arguments to this function. The real test as to whether it works or not is if AWS accepts it and it behaves as expected. I wouldn't trust their offline emulator for this. Try it for real in a deployed function and see what the results are.

If it works as expected in AWS, then I can revisit the grammar for the asl-path-validator and support inline JSON but I'm skeptical that this is something AWS will accept.

**Comment by remy90** [comment]
Not sure why but they don't, it appears!
